src/Statement/Filter/DataTypeStatementFilter.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Filter;
+
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookup;
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookupException;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementFilter;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class DataTypeStatementFilter implements StatementFilter {
+
+	/**
+	 * @var PropertyDataTypeLookup
+	 */
+	private $dataTypeLookup;
+
+	/**
+	 * @var string[]
+	 */
+	private $dataTypes;
+
+	/**
+	 * @param PropertyDataTypeLookup $dataTypeLookup
+	 * @param string[]|string $dataTypes One or more data type identifiers.
+	 */
+	public function __construct( PropertyDataTypeLookup $dataTypeLookup, $dataTypes ) {
+		$this->dataTypeLookup = $dataTypeLookup;
+		$this->dataTypes = (array)$dataTypes;
```

Comment:
ack, that would be better. Or, as Jeroen suggested below, use http://php.net/manual/en/class.splobjectstorage.php.